### PR TITLE
python3Packages.angrcli: init at 1.1.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python modules to allow easier interactive use of angr

https://github.com/fmagin/angr-cli

Related to #81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 126515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.angrcli</li>
    <li>python39Packages.angrcli</li>
  </ul>
</details>

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -407,6 +407,8 @@ in {
 
   angr = callPackage ../development/python-modules/angr { };
 
+  angrcli = callPackage ../development/python-modules/angrcli { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:410]:
```suggestion
  angrcli = callPackage ../development/python-modules/angrcli {
    inherit (pkgs) coreutils;
  };
```

File: pkgs/development/python-modules/angrcli/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  };
+
+  propagatedBuildInputs = [
+    angr
+    cmd2
+    pygments
+  ];
+
+  checkInputs = [
+    pkgs.coreutils
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/angrcli/default.nix:31]:
```suggestion
    coreutils
```

File: pkgs/development/python-modules/angrcli/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    pytestCheckHook
+  ];
+
+  postPatch = ''
+    # Version mismatch, https://github.com/fmagin/angr-cli/pull/11
+    substituteInPlace setup.py \
+      --replace "version='1.1.0'," "version='${version}',"
+    substituteInPlace tests/test_derefs.py \
+      --replace "/bin/ls" "${pkgs.coreutils}/bin/ls"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/angrcli/default.nix:40]:
```suggestion
      --replace "/bin/ls" "${coreutils}/bin/ls"
```

File: pkgs/development/python-modules/angrcli/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
+{ lib
+, angr
+, buildPythonPackage
+, cmd2
+, fetchFromGitHub
+, pkgs
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/angrcli/default.nix:6]:
```suggestion
, coreutils
```

**Review by SuperSandro2000 — DISMISSED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126515` at fbf7ca73 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/c47ad975282af7e777b573cfd44ef611">python38Packages.angrcli</a></li>
    <li><a href="https://gist.github.com/e5b33fcfe8cf3cd7ad36f61a70289186">python39Packages.angrcli</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/top-level/python-packages.nix:410]: Thanks, change.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 126515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.angrcli</li>
    <li>python39Packages.angrcli</li>
  </ul>
</details>

**Review by jonringer — COMMENTED** [review]
I don't see a cli command export..
```
$ tree -L 2 ./results/python38Packages.angrcli/
./results/python38Packages.angrcli/
├── lib
│   └── python3.8
└── nix-support
    └── propagated-build-inputs
```

Can we get upstream to export a proper `entry_points`?

**Comment by fabaff** [comment]
> Can we get upstream to export a proper entry_points?

I'm not sure how active upstream is as there as not much action lately. I think it's definitely worth looking into the matter.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 126515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.angrcli</li>
    <li>python39Packages.angrcli</li>
  </ul>
</details>

**Pull request merged**